### ISSUE TITLE
Training script

### ISSUE THREAD

**Issue by HaishuoFang** [Chen-GX/IterResearch#2]
Thanks for your public repo. I was wondering if there is any plan to release the training script? Or could you explain how you calculate the advantages for different rollouts? Do you consider all steps in all rollouts in a group or is there any other methods? Thank you in advance!

**Comment by Chen-GX** [comment]
Thanks for your interest in our work. I apologize for the fact that we do not have any plan for releasing the training script, due to some constraints.

For the advantage calculation, I could tell more detailed exps after we finish this paper.

Besides the calculation in our paper (where we use discounted reward and calculate the advs by putting all the steps in one group, just as you mentioned), we also have attempted the following method:
(Note that, these method have discarded the discounted reward, 'cause we wanna focus on the advantage calculation. That means we will share the same reward between intermediate step and final step. )

1. Advantage calculatoin with all steps in one group, which is unstable during training. I this this is 'cause the intermediate steps are more than the final steps, which means the adv will be significantly affected by the number of the steps. And we observe that the negative trajs always have more steps than the positive trajs.
2. Separated goup between the imtermediate and the final steps. And we calculate the adv independently.
3. Only calculating the advantage in the final steps and sharing its adv among the intermediate steps in the same traj.

Our conclusion is that method 2/3 are better and stable than 1.

BTW, I think we should use PPO instead of GRPO in the paradigm of IterResearch due to the difficulty of advantage calculation, where the value model will give an estimated value in each step, just like PRM.

That's all for my experiences and suggestions beyond the paper itself. I hope these could help you a lot.

Feel free for any question. I think iterResearch is an interesting paradigm and has a lot of potential with its capability of iteraction scaling and the evolved report mechanism.
